02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -170,26 +170,158 @@ Here's what has been added to the above since then:
 
 ## Why is PureScript's Documentation Lacking and How Do We Improve It?
 
-- Philosophy of the language:
-    - avoid (success at all costs)
-    - wrong intepretation:
-    - right interpretation
-- Current priority of core contributors: adding powerful features via breaking changes
-    - wrong interpretation: they don't care
-    - right interpretation: want to improve language
-    - Why a `v1.0` hasn't been released
-- Core Contributors are spread thin, so they cannot respond quickly to things
-    - The '75 libraries' problem
-    - Entrusting library maintenance to others and supporting them (e.g. the `event-stream` incident)
-    - Just like reviewing code, reviewing documentation takes a lot of thought
-    - Hard to keep documentation momentum
-    - Which libraries are 'core libraries'?
-- No clearly-defined mutually-held community-wide goal
-    - A desire to help exists, but no one is coordinating resources strategically
-- What exactly are FP's "Best Practices?"
-    - They are not well-defined, are assumed, or are habits
-    - Knowledge silos exist, so it's hard to extract that info
-- Slack-based discussions do not persist
+These are the reasons. Each wil be covered more in-depth following this:
+
+- PureScript is not _currently_ trying to be the next "mainstream language".
+- Core contributors are spread thin: limited time with too much responsibility
+- There is no roadmap that coordinates efforts
+- It's difficult to pin down and define what are Functional Programming's "best practices"
+- Questions answered on Slack do not persist, so they get re-asked and re-answered
+
+### PureScript is not _currently_ trying to be the next "mainstream language"
+
+Many have wrongly assumed that PureScript is trying to replace [insert your favorite web language here]. That is not the case. Each language has its trade-offs that make it better for some and worse for others.
+
+Rather, PureScript is _currently_ following the motto of "**avoid (success at all costs)**".
```

Comment:
> Actually, this motto is more confusing that illuminating. 

Agreed. I'm not even sure whether we are understanding it correctly ourselves. When I read it for the first time, it didn't make any sense. I think it's stating something in the negative (we are against X) instead of stating something in the positive (we are for Y). The former doesn't actually tell me what you value whereas the other does.
Also, including `avoid (success at all costs)` here might be a bad thing. I believe people have negative emotional connotations when they read it. It might imply "You're stupid" to people who tried learning FP beforehand and failed or gave up.

> If we make this change, the paragraphs which follow will need to be changed. Currently they seem to be explaining the motto's meaning, so we could change them to be explanations of why this motto is beneficial. While I would guess the benefits were intended to primarily benefit project maintainers, I believe there are also benefits for users of the language and tools.

I agree with changing the following paragraphs. This phrase is itself a form of jargon. We should invert the order to explain the interpretation first before 'naming' that interpretation:
- Values that FP priortize
- Values that non-FP prioritizes
- Name it using Jargony Phrase (or, more likely, some non-jargony phrase that's still clear)

Reply:
> Perhaps we should just use our interpretation instead, which might be "Avoid satisfying the majority of your users when there it increases maintenance cost" or a more compact form of this.

I think it's more along the lines of "avoid 'dumbing down the language / de-powering the language' to make it more convenient / understandable for people who don't know the foundations behind it"

I don't think maintenance has anything to do with it.

Reply:
So, I did some googling. This one post summarizes it clearly: https://news.ycombinator.com/item?id=12056169

> The Haskell motto is "avoid success at all costs", usually clarified to "avoid 'success at all costs'".
> 
> In other words, "success" (popularity, widespread use in industry, teaching, etc.) is not a justification for "costs" such as limiting expressiveness, making things less safe, restricting research possibilities, etc. Haskell would prefer to be powerful, safe, efficient, obscure and niche; rather than popular, industry-standard, widely-known, highly compatible, unsafe, insecure, inefficient and restricted (besides, we have loads of languages of this sort!).
> 
> Crippling the language to make things easier for new users is not an option. However, there are many ways the language could be made easier without impacting any core values. For example, there are long-standing issues with strings, records, error messages, dependency handling, etc. which people are working on.
> 
> Also, for those who do want to learn Haskell, I've found the community to be very welcoming and tolerant of newbies. Just ignore some of the more math-heavy blog posts (for now, at least), or your head might explode :P

Reply:
In the latest changes, I summarized this quote and deleted the rest of what I wrote.